contribute/design/1.x/ndm/20190705-ndm-disk-identification.md

```diff
@@ -0,0 +1,196 @@
+    	  even if the device path changes
+    2. Check if the disk has a Filesystem label available on it. This is in cases where the 
+       complete disk is formatted with a filesystem without a partition table. If filesystem
+       label is available we use it for generating the UUID.
+    3. If GPT and FS label is not available on the disk, we create a GPT partition table on the disk,
+       and then create a single partition which spans the complete disk. This new partition will
+       be consumed by the users. This will help to identify a virtual disk for movements across
+       the cluster also.
+
+Data Cleanup:
+NDM takes care of data clean up, after a BD is released from a BDC. i.e. a complete
+wipe of the disk will be done `wipefs -fa`. Since in case of NDM created GPT labels, 
+only partition is being wiped, it won't cause the labels to be removed.
+
+##### Workflow
+```
++-----------+
+|UID = WWN +|
+|    Model +|
+|   Serial +|
+|   Vendor  |
+|           |
++-----------+
+     |
+     |
++----v-------------+        +-----------+      +-----------+      +-------------------+
+|    If            |        |  Check    |      |  Check    |      | Create GPT        |
+|ID_TYPE is empty  |   Yes  | GPT label |  No  | FS label  |  No  | partition         |
+|    or ID_MODEL is+------->+           +----->+           +----->+ table and         |
+|Virtual/Ephemeral |        |           |      |           |      | one partition     |
+|Disk and empty WWN|        |           |      |           |      | spanning the disk |
++------------------+        +-----------+      +-----------+      +-------------------+
+     |                        |                     |                  |
+     |No                      |                     |                  |
+     v                        |Yes                  |Yes               |
++-----------+                 |                     |                  |
+| Use       |                 |                     |                  |
+| generated |<----------------v---------------------v------------------+
+| UID  or   |
+|  Label    |
+|           |
++-----------+
+```
+
+## Graduation Criteria
+
+- NDM should be able to uniquely identify the disks, across reboots, across different
+  SCSI ports and anywhere within the cluster. The unique identification can be marked
+  successful, if NDM detects the disk, and the user is able to retrieve his data
+  from the disk
+
+## Implementation History
+
+- Owner acceptance of `Summary` and `Motivation` sections - YYYYMMDD
+- Agreement on `Proposal` section - YYYYMMDD
+- Date implementation started - YYYYMMDD
+- First OpenEBS release where an initial version of this OEP was available - YYYYMMDD
+- Version of OpenEBS where this OEP graduated to general availability - YYYYMMDD
+- If this OEP was retired or superseded - YYYYMMDD
+
+## Drawbacks
+
+- Do not have a mechanism to identify disks that are attached in multipath mode.
```

Comment:
isn't this the case even with current implementation?
How about limitation that north bound user of NDM will not be able to make any more partitions?

Reply:
Added this into list of limitations